consensus/ibft/signer/key_manager.go

```diff
@@ -0,0 +1,21 @@
+package signer
+
+import (
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/0xPolygon/polygon-edge/validators"
+)
+
+// KeyManager is a delegated object to sign data
+type KeyManager interface {
+	Type() validators.ValidatorType
+	Address() types.Address
+	NewEmptyValidatorSet() validators.Validators
+	NewEmptyCommittedSeal() Sealer
+	SignSeal([]byte) ([]byte, error)
+	SignCommittedSeal([]byte) ([]byte, error)
+	Ecrecover(sig []byte, digest []byte) (types.Address, error)
+	GenerateCommittedSeals(map[types.Address][]byte, *IstanbulExtra) (Sealer, error)
+	VerifyCommittedSeal(validators.Validators, types.Address, []byte, []byte) error
+	VerifyCommittedSeals(Sealer, []byte, validators.Validators) (int, error)
+	SignIBFTMessage(msg []byte) ([]byte, error)
+}
```

Comment:
This interface does way too much

cc @dbrajovic

Reply:
The same reason above. If you want to split this interface, please tell me your suggestion

Reply:
@zivkovicmilos ping